Automattic/wp-calypso
# Stats: create redux version of stats-list.

As part of migrating from the legacy `stats-list` to utilizing the global state tree across stats, this branch introduces a new version of the `<StatsModule />` component.  The `<StatsConnectedModule />` will help ease the transition of all legacy `<StatsModule />` usage to utilize Redux.

I attempted to cobble together support for redux + `stats-list` within the existing `<StatsModule />` but after fighting backwards compatibility - I felt creating a new temporary module to use during the transition was the best approach.

The first stats module to use this new `Connected` component is the "Publicize" module on the Insights page.  This is a very basic module with no row actions or summary page support - so it seemed like a great place to start.

<img width="366" alt="stats_ _wordpress_com" src="https://cloud.githubusercontent.com/assets/22080/16173382/93163df2-3554-11e6-8f2b-95cc1e468165.png">


__To Test__
- Visit a site insights page
- Validate the Publicize module loads as expected, and shows the same data as seen in production

__Implementation Notes__
Additionally this branch introduces a new selector `getSiteStatsParsedData`.  Akin to the legacy `stats-list`, this selector will call the appropriate utility method in `state/stats/lists/utils` for the given `statType`.  In this branch you can see tests added for this new selector along with tests for parsing `statsPublicize` data.

After all existing stats modules are updated to use `<StatsConnectedModule />` my plan will be to delete the old version of `<StatsModule />` and let the connected version become the default use case.  This approach will allow us to do smaller/focused PRs to migrate all stats modules over to redux.

Test live: https://calypso.live/?branch=update/stats/publicize
